From 519472e2ff8c3c28448e3be4782bd298c089cb7e Mon Sep 17 00:00:00 2001
From: yeflin01 <yefeng.lin@arm.com>
Date: Fri, 3 Feb 2023 17:12:13 +0800
Subject: [PATCH] extended  wrk2 eventLoop size to run over vpp

Wrk2 will raise an error, if the socket file descriptor is larger than
the eventLoop size. This can occur when running wrk2 over VPP, as the
file descriptor range of the socket is larger than that of running over
the Kernel. To ensure wrk2 can run normally over VPP, it is necessary to
extend the eventLoop size.

Signed-off-by: Yefeng Lin <yefeng.lin@arm.com>
---
 src/wrk.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wrk.c b/src/wrk.c
index 28ed17e..113e2b0 100644
--- a/src/wrk.c
+++ b/src/wrk.c
@@ -133,7 +133,7 @@ int main(int argc, char **argv) {
     uint64_t thread_init = time_us();
     for (uint64_t i = 0; i < cfg.threads; i++) {
         thread *t = &threads[i];
-        t->loop        = aeCreateEventLoop(10 + cfg.connections * 3);
+        t->loop        = aeCreateEventLoop(31 + cfg.connections * 3);
         t->connections = connections;
         t->throughput = throughput;
         t->reconnect_all = NULL;
-- 
2.17.1

